jax/experimental/jax2tf/tests/primitive_harness.py

```diff
@@ -95,6 +95,12 @@ class CustomArg(NamedTuple):
 
 ArgDescriptor = Union[RandArg, StaticArg, CustomArg, Any]
 
+# We sanitize test names to match pytest "-k" command-line option
+# notion of identifier.
+kSanitizeNameTranslateTable = str.maketrans({
+  " ": "", "(": "[", ")": "]", "{": "[", "}": "]", "<": "[", ">": "]",
```

Comment:
I think these aren't enough. `[` and `]` are the main problematic characters because `-k` takes passes its argument to `fnmatchcase`, and `fnmatchcase` treats `[` specially:

https://docs.python.org/3/library/fnmatch.html

I think the logic in the `unittest` library that matches is this:
https://github.com/python/cpython/blob/fc130c47daa715d60d8925c478a96d5083e47b6a/Lib/unittest/loader.py#L210

The CPython documentation seems misleading, since it says only `*` is special. But reading the code that's clearly false: https://docs.python.org/3/library/unittest.html#cmdoption-unittest-k